tryiiif/home/__init__.py

```diff
@@ -30,7 +30,13 @@ def index():
         manifest = make_manifest(uid, url, b64url, name, info['height'],
                                  info['width'])
         rc.conn.set(uid, json.dumps(manifest))
-        return redirect(url_for('viewers.viewer', viewer='uv', uid=uid))
+
```

Comment:
Yeah, I considered that but I was wondering if we wanted to avoid people being able to enter whatever they wanted as a value. I know it’s not sql injection, but I just tend to positively test for stuff that users can enter into a form. Am I being silly?

From: Mike Graves <notifications@github.com<mailto:notifications@github.com>>
Reply-To: MITLibraries/tryiiif <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Monday, May 23, 2016 at 3:05 PM
To: MITLibraries/tryiiif <tryiiif@noreply.github.com<mailto:tryiiif@noreply.github.com>>
Cc: Solh Zendeh <szendeh@mit.edu<mailto:szendeh@mit.edu>>, State change <state_change@noreply.github.com<mailto:state_change@noreply.github.com>>
Subject: Re: [MITLibraries/tryiiif] adding mirador viewer (#8)


In tryiiif/home/__init__.py<https://github.com/MITLibraries/tryiiif/pull/8#discussion_r64268039>:

> @@ -30,7 +30,13 @@ def index():
>          manifest = make_manifest(uid, url, b64url, name, info['height'],
>                                   info['width'])
>          rc.conn.set(uid, json.dumps(manifest))
> -        return redirect(url_for('viewers.viewer', viewer='uv', uid=uid))
> +


In order to avoid a proliferation of ifs, I'd recommend something like the following:

return redirect(url_for('viewers.viewer', viewer=request.form['submit'], uid=uid))

—
You are receiving this because you modified the open/close state.
Reply to this email directly or view it on GitHub<https://github.com/MITLibraries/tryiiif/pull/8/files/e34e2c8f18d98d34724f541fa1aad70e66e5dac7..a7a4810a4012ff681faba9fcbcf8599725db7a14#r64268039>